static/js/message_list_view.js

```diff
@@ -486,6 +479,35 @@ MessageListView.prototype = {
             });
         }
 
+        row.find('.user-mention').each(function () {
+            var user_id = get_user_id_for_mention_button(this);
+            if (user_id && user_id !== "*" && !$(this).find(".highlight").length) {
+                // If it's a mention of a specific user, edit the
+                // mention text to show the user's current name,
+                // assuming that you're not searching for text
+                // inside the highlight.
+                $(this).text("@" + people.get_person_from_user_id(user_id).full_name);
+            }
+        });
```

Comment:
Let's do this piece as its own commit; also, I think it might be more correc to just remove the `row.hasClass('mention')` if statement, rather than moving this block out.  And then we can not duplicate the `user-group-mention` block too...